src/js/locks.js

```diff
@@ -52,7 +52,8 @@ define(function (require, exports, module) {
         JS_STYLE: "jsStyle",
         JS_LIBRARIES: "jsLibraries",
         CC_LIBRARIES: "ccLibraries",
-        OS_CLIPBOARD: "osClipboard"
+        OS_CLIPBOARD: "osClipboard",
+        PS_GENERATOR: "psGenerator"
```

Comment:
Probably no need for a `PS_` prefix here? That is, probably no need to add it to `ALL_PS_LOCKS`. But, might want to add it to `ALL_NATIVE_LOCKS`?